pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1317,11 +1318,24 @@ func (vs *VSphere) CreateVolume(volumeOptions *vclib.VolumeOptions) (canonicalVo
 				}
 			}
 		}
-		ds, err := dc.GetDatastoreByName(ctx, datastore)
+
+		// When zone or storage profile is not specified in storage class, datastoreInfo is still not determined
+		if datastoreInfo == nil {
```

Comment:
As per the existing if else blocks above, looks like datastoreInfo is not expected to be nil at this point.

Reply:
Removed this check and raised an error here if datastoreInfo is still nil.